app/controllers/gestionnaires/dossiers_controller.rb

```diff
@@ -109,8 +109,12 @@ def repasser_en_instruction
       if dossier.en_instruction?
         flash.notice = 'Le dossier est déjà en instruction.'
       else
-        flash.notice = "Le dossier #{dossier.id} a été repassé en instruction."
-        dossier.repasser_en_instruction!(current_gestionnaire)
+        if dossier.accepte? && !administration_signed_in?
```

Comment:
J’ai un peu de mal à comprendre l’état précédent du code. Si je comprends bien: les instructeurs pouvaient repasser un dossier en instruction, mais ça échouait si le dossier était accepté parce que la transition `from: :accepte, to: :en_instruction` n’existait pas dans dossier.rb. OK.

Si je comprends bien, maintenant on gère ce cas dans le controller directement, et on laisse la permission au superadmins de le faire quand même. Est-ce que c’est pertinent de gérer le cas des superadmins ici? Est-ce qu’on ne pourrait pas faire seulement:
```suggestion
        if dossier.accepte?
```
et laisser les superadmins faire ce qu’ils veulent dans `/manager`, mais pas ici?

Reply:
pour le premier paragraphe c'est ça. Pour le second je veux éviter les cas qu'on a où des instructeurs ont quand même accès au bouton pour passer en instruction (il y a des configurations relous où ça arrive, d'où pas mal de code défensif dans ce controlleur).

Le problème c'est surtout que la logique de sécurité est au niveau du controlleur, alors que les règles de transition sont dans le modèle. C'est du au fait qu'on a pas accès aux méthodes `xxx_signed_in?` dans le modèle, d'où la nécessité de cette duplication… Je ne sais pas si pundit résoudrait le problème.

Reply:
Les admins (normaux) n’ont jamais le droit de repasser en instruction un dossier accepté, non? En fait je propose de retirer la deuxième condition (`&& !administration_signed_in?`) qui ne fait qu’autoriser les superadmins à repasser en instruction.

Reply:
Justement :) dans le modèle on autorise désormais la transition accepté->instruction sans condition sur l'utilisateur (vu qu'on y a pas accès).
Donc sans cette condition `&& !administration_signed_in?`, dans les situations où il arrive qu'il y ait des transitions non voulues (cf les quelques "invalid transition" dans sentry), on risque de permettre aux instructeurs de le faire.
Par ailleurs, sans cette condition, des administrateurs malins peuvent également forger les requêtes nécessaires pour faire cette transition.

Reply:
Aaah ok, j’avais beau relire les trois lignes, je comprenais l’inverse 😕.
Ma confusion venait de la proposition que je veux faire: ne pas gérer le cas des superadmins ici.
C’est à dire: toujours refuser le passage accepté -> en instruction dans `/gestionnaires/dossiers_controllers.rb`, le permettre uniquement dans `/manager/dossiers_controllers.rb`.

Reply:
C'est ça :)